src/collection.js

```diff
@@ -616,11 +615,12 @@ export default class Collection {
    */
   gatherLocalChanges() {
     let _toDelete;
-    return listStatuses(this.db, ["created", "updated", "deleted"])
-      .then(([created, updated, deleted]) => {
-        _toDelete = deleted;
+    return this.list({filters: {_status: ["created", "updated", "deleted"]}},
+                     {includeDeleted: true})
+      .then((results) => {
+        _toDelete = results.data.filter(r => r._status === "deleted");
         // Encode unsynced records.
-        const unsyncRecords = created.concat(updated);
+        const unsyncRecords = results.data.filter(r => r._status !== "deleted");
```

Comment:
This is probably nitpicking, but I think this particular case I would prefer the version with two calls to `list()`. 

```js
return Promise.all([
   this.list({filters: {_status: ["created", "updated"]}}),
   this.list({filters: {_status: "deleted"}}),
)
.then(([unsynced, deleted]) => {
   ...
```
```